agent/loop.py

```
@@ -511,8 +563,242 @@ async def _dispatch_tool(
+                    f"Failed steps: {[s.get('description') for s in failed]}"
+                ),
+            },
+        ]
+        try:
+            raw = await self._chat_json(judge_model, messages)
+            verdict = raw.get("verdict", "APPROVED_WITH_CONDITIONS")
+            if verdict == "BLOCKED":
+                log.warning("Judge BLOCKED session %s: %s", session_id, raw.get("notes", ""))
+            self._log_event(session_id, "assistant_message", {"judge": raw})
+            return raw
+        except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)